gem/lib/pupperware/spec_helper.rb

```diff
@@ -149,6 +164,11 @@ def emit_log(container)
     STDOUT.puts(logs)
   end
 
+  def teardown_container(container)
+    STDOUT.puts("Tearing down test container")
+    run_command("docker container rm --force #{container}")
```

Comment:
Should we make `teardown_cluster` (`docker container kill #{id}`) call this... or should we change this to match the semantics above?

IIRC, the reason we do `kill` and not `rm` above is because `rm` deletes the logs... and at least in the other suites, we stop containers prior to emitting logs (then let the cleanup scripts take care of any additional cleanup).

Thoughts?

Reply:
Doesn't the `docker-compose down` in `teardown_cluster` remove the containers? I'm not seeing anywhere other than in the `teardown_cluster` that we're stopping containers, but I might be missing something.

I was mostly doing the `rm --force` since it would stop and rm the containers, and calling `emit_log` right before `teardown_container`, which looks like the same order as in pdb and puppetserver.

So, yeah, might make sense to update `teardown_cluster` to use `teardown_container`.

Reply:
Ah yeah thanks... we're doing `emit_logs` first already!